couchdb/src/test/java/org/hibernate/ogm/datastore/couchdb/test/crud/EntityWithByteArrayTest.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.couchdb.test.crud;
+
+import java.util.Arrays;
+import java.util.Random;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.annotations.GenericGenerator;
+import org.hibernate.ogm.utils.OgmTestCase;
+import org.hibernate.ogm.utils.TestForIssue;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * @author Hardy Ferentschik
+ */
+public class EntityWithByteArrayTest extends OgmTestCase {
```

Comment:
> IMO it makes sense to handle and fix it generically if we find out it applies to all stores actually.

Sure,  I am not saying that it should not be fixed for the other stores, but I don't like modifying or changing the scope of a specific user issue.

This could for example be merged and fix the specific user issue. If I increase the scope of the issue by trying to fix it for all datastores, it might not get ready for the next release. Or maybe we say that the support for byte[] has a lower priority than other issue and we should do something else first. In this case there is again no immediate fix for this particular issue even so it is ready.

Supporting byte[] for all data stores is a different beast to supporting it for CouchDB only and addressing this particular issue. 

Also, the whole idea of not duplicating tests is hard to keep up anyways, since it would require at all times to know what and where things are tested.